### PR TITLE
fix(products): fix stale product description

### PR DESCRIPTION
**Pull request by beasync**
Contains the following changes:

- Fix stale product description
- Fix `AccessTokenNotAvailableException` handling in `EsquioHttpClient`

TL;DR

Fix stale product description

After adding a new product, the DOM structure of the previous cards is reused, causing the description of the product that previously occupied its position in the grid to appear when editing the description of the new product.

Adding the `@key` property forces to discard the previous DOM structure generating a new structure for the newly added product, solving the problem.

Before:

stale_product_cards_before

After:

stale_product_cards_after

Fix `AccessTokenNotAvailableException` handling in `EsquioHttpClient`

Addition of the `AccessTokenNotAvailableException` exception handling as specified in the [documentation](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/blazor/webassembly/additional-scenarios.md#attach-tokens-to-outgoing-requests), redirecting the user without an available access token to the authentication flow.

### CODE REVIEW

**Comment by unaizorrilla** [comment]
Merged! thanks, I think there is some issue on github because the check failing is not sense!
